armsrc/lfsampling.c

```diff
@@ -424,6 +425,123 @@ uint32_t SampleLF(bool verbose, uint32_t sample_size, bool ledcontrol) {
     BigBuf_Clear_ext(false);
     return ReadLF(true, verbose, sample_size, ledcontrol);
 }
+
+/**
+ * Do LF sampling and send samples to the USB
+ * 
+ * Uses parameters in config. Only bits_per_sample = 8 is working now
+ * 
+ * @param reader_field - true for reading tags, false for sniffing
+ * @return sampling result
+**/
+int ReadLF_realtime(bool reader_field) {
+    // parameters from config and constants
+    const uint8_t bits_per_sample = config.bits_per_sample;
+    const int16_t trigger_threshold = config.trigger_threshold;
+    int32_t samples_to_skip = config.samples_to_skip;
+    const uint8_t decimation = config.decimation;
+
+    uint32_t sample_buffer_len = 64;
+    const int8_t size_threshold_table[9] = {0, 64, 64, 60, 64, 60, 60, 56, 64};
+    const int8_t size_threshold = size_threshold_table[bits_per_sample];
+
+    // DoAcquisition() start
+    uint8_t last_byte = 0;
+    uint8_t curr_byte = 0;
+    int return_value = PM3_SUCCESS;
+
+    initSampleBuffer(&sample_buffer_len);
```

Comment:
May I suggest moving the local declaration of `sample_buffer_len` to be just above the call to `initSampleBuffer()`.

There's still a duplicated hard-coded magic number (64) used herein.  I don't know what style is preferred (obviously, Iceman's opinion will rule), so here's two strawman examples:

<details><summary>Strawman Alpha</summary><p/>

```C
static const uint32_t sample_buffer_len = 64;
uint32_t actual_buffer_len = sample_buffer_len;
initSampleBuffer(&actual_buffer_len);
if (actual_buffer_len != sample_buffer_len) { //...
```
</details> 

<details><summary>Strawman Omega</summary><p/>

```C
static const uint32_t SAMPLE_BUFFER_LENGTH = 64;
uint32_t sample_buffer_len = SAMPLE_BUFFER_LENGTH;
initSampleBuffer(&sample_buffer_len);
if (SAMPLE_BUFFER_LENGTH != sample_buffer_len) { //...
```

</details>

Reply:
The magic number there can be `AT91C_EP_IN_SIZE`, but it's defined in `common_arm/usb_cdc.c` so I need to export it.

Reply:
Fixed